rollbar/contrib/fastapi/utils.py

```diff
@@ -21,14 +21,50 @@ def __init__(self, version, reason=''):
         return super().__init__(err_msg)
 
 
+def is_current_version_higher_or_equal(current_version, min_version):
+    if current_version == min_version:
+        return True
+
+    for current_as_string, min_as_string in zip(
+        current_version.split('.'),
+        min_version.split('.'),
+    ):
+        if current_as_string == min_as_string:
+            continue
+
+        try:
+            current_as_int = int(current_as_string)
+        except ValueError:
+            current_as_int = None
+
+        try:
+            min_as_int = int(min_as_string)
+        except ValueError:
+            min_as_int = None
+
+        if current_as_int is None or min_as_int is None:
+            # If one of the parts fails the int conversion, compare as string
+            return current_as_string > min_as_string
+        else:
+            if current_as_int == min_as_int:
+                continue
+            return current_as_int > min_as_int
+
+    # Somehow the comparison didn't properly finish - defaulting to False
+    return False
```

Comment:
Per [PEP 440](https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions), this could probably be simplified.

FastAPI's version needs to be PEP440 compliant, irrespective of SemVer's spec.

Unfortunately that means fully-complaint checking requires things like epoch prefixes and `-dev` sorting before `-beta`.

Fortunately however, I think we can outright ignore everything except the release segment (i.e., ignore `-alphaN`, `-betaN`, -`devN`, `preN`, etc.) for the purposes of compatibility checking. That is, unless we *want* to say the min version of FastAPI supported by Rollbar is something like `0.104.0-beta2` and `0.104.0-beta1` is not allowed. That seems very unlikely to be useful.

```suggestion
def is_current_version_higher_or_equal(current_version, min_version):
    # We can ignore everything except the release segment
    current = tuple(map(int, re.findall(r"(?:^|[.])([0-9]+)", current_version)))
    minimum = tuple(map(int, re.findall(r"(?:^|[.])([0-9]+)", min_version)))
    return current >= minimum
```

Reply:
@johnchristopherjones I agree that only trying to compare the release segment seems reasonable. 

However, that regex also captures pre-release and dev segments if they include a `.[0-9]`. This does mean that `1.0.0-alpha.1` would become the tuple `(1, 0, 0, 1)`. I am not great at regex. So, after a few failed attempts to solve this, I resorted to solving it in Python.